subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyManagementResultsAsInputsIntegrationTest.groovy

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve
+
+import groovy.test.NotYetImplemented
+import org.gradle.api.internal.artifacts.DefaultModuleIdentifier
+import org.gradle.api.internal.artifacts.DefaultModuleVersionIdentifier
+import org.gradle.api.internal.artifacts.result.DefaultResolvedVariantResult
+import org.gradle.api.internal.attributes.ImmutableAttributesFactory
+import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
+import org.gradle.integtests.fixtures.daemon.DaemonLogsAnalyzer
+import org.gradle.internal.Describables
+import org.gradle.internal.component.external.model.DefaultModuleComponentArtifactIdentifier
+import org.gradle.internal.component.external.model.DefaultModuleComponentIdentifier
+import org.gradle.internal.component.external.model.ImmutableCapability
+import org.gradle.test.fixtures.file.TestFile
+import spock.lang.Issue
+
+class DependencyManagementResultsAsInputsIntegrationTest extends AbstractHttpDependencyResolutionTest {
```

Comment:
It would be good to verify that the artifact metadata forms part of the task inputs when a real `ResolvedArtifactResult` (rather than one created with the internal constructor) is used as a task input, eg adding or changing a variant attribute in the library project will cause the consuming task to run again, even though the file locations and file contents have not changed.

Reply:
Done for project, included and external dependencies.